Uplift/Models/Gym.swift

```diff
@@ -86,8 +87,9 @@ struct Gym {
         })
 
         facilities = gymData.facilities.compactMap {
-            guard let facility = $0 else { return nil }
-            return Facility(facilityData: facility)
+            guard let facility = $0,
+                let facilityType = FacilityType(rawValue: facility.name) else { return nil }
+            return Facility(facilityData: facility, facilityType: facilityType)
```

Comment:
There's some repeat of code in this `init` function and the other one (specifically lines 42/61 and 73/89 for the respective functions); can we mitigate that?

Reply:
theres no guarantee that `facility.name` will successfully convert to a `FacilityType` enum so I have to unwrap it outside the `init` for both